src/rgw/rgw_auth_keystone.cc

```diff
@@ -167,8 +166,14 @@ TokenEngine::get_acl_strategy(const TokenEngine::token_envelope_t& token) const
     make_spec_item("*", user_name),
   };
 
+  std::list<rgw::keystone::TokenEnvelope::Role> token_roles;
+  for (const auto& r : token.roles) {
+    token_roles.insert(token_roles.end(),
+        rgw::keystone::TokenEnvelope::Role(r));
+  }
```

Comment:
the `token` being passed into `get_acl_strategy()` lives on the stack in `TokenEngine::authenticate()`, so we can't capture it by reference in the lambda - this lambda gets stored in the `IdentityApplier` for use later in the request (long after `authenticate()`)

you're making a copy of `token.roles` here in the local variable `token_roles`, and this local copy can't be captured by reference for the same reason

so we *do* need to capture the roles by-value, but we could avoid this extra local copy into `token_roles` by copying directly into the lambda:
```
   return [allowed_items, token_roles=token.roles](const rgw::auth::Identity::aclspec_t& aclspec) {
```

Reply:
thanks, casey

Reply:
The documentation that I have says that a capture with initializer is a feature of C++14 and later. If you say it's okay to use, I'm only happy to use it.